### PR TITLE
refactor: handleNewBooking #4

### PR DESCRIPTION
**Pull request by Udit-takkar**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

Continuing refactor of handleNewBooking 
- https://github.com/calcom/cal.com/pull/15532
- https://github.com/calcom/cal.com/pull/15590
- https://github.com/calcom/cal.com/pull/15612

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

### CODE REVIEW

File: packages/features/bookings/lib/handleNewBooking/checkBookingAndDurationLimits.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/checkBookingAndDurationLimits.ts:15]: check both booking and duration limits.

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking.ts:252]: moved validation code

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking.ts:175]: it's just more simpler to read.

File: packages/features/bookings/lib/handleNewBooking/getCustomInputsResponses.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/getCustomInputsResponses.ts:48]: refactored this function

File: packages/features/bookings/lib/handleNewBooking/getCustomInputsResponses.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking/getCustomInputsResponses.ts:37]: use `.reduce` method instead of `for` loop

File: packages/features/bookings/lib/handleNewBooking.ts
Comment (Udit-takkar) [inline packages/features/bookings/lib/handleNewBooking.ts:296]: I have used let here because it `originalRescheduledBooking` is re assigned

**Comment by Udit-takkar** [comment]
To be merged after "booking with phone number"

**Review by joeauyeung — DISMISSED** [review]
Great job refactoring this huge file @Udit-takkar.

Just a heads up I tried creating a booking with multiple durations and choosing the non-default duration and got an invalid length error.

**Comment by Udit-takkar** [comment]
> It looks like this function is doing two things either getting the rescheduled booking or getting the seated booking. I think if we're going to abstract this logic it should be split into two functions that each have a single purpose.

Thanks for the suggestion. It's done now.

**Review by joeauyeung — DISMISSED** [review]
Feedback is addressed. Great refactoring @Udit-takkar

**Comment by Udit-takkar** [comment]
@joeauyeung fixed merge conflicts

**Review by joeauyeung — APPROVED** [review]
Approving again after fixing merge conflicts

**Pull request merged**
